model/internal/data/optional.go

```diff
@@ -0,0 +1,71 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package data // import "go.opentelemetry.io/collector/model/internal/data"
+
+import (
+	encoding_binary "encoding/binary"
+	"math"
+)
+
+type OptionalDouble struct {
```

Comment:
This is "OptionalFixed64" correct (is not the int64 from proto is the fixed64)?

Reply:
Correct, this is a `fixed64`  field.